ansible-scylla-node/tasks/ssl.yml

```diff
@@ -1,20 +1,33 @@
 ---
 - name: generate a self-signed CA and use it for node certs
   block:
+    - name: Check CA folder exists
+      stat:
+        path: ./ssl/ca
+      register: ca_folder_details
+
     - name: Create dir for the CA
       file:
         path: ./ssl/ca
         state: directory
+      when: ca_folder_details.stat.isdir == False
+
+    - name: Check CA pem exists
+      stat:
+        path: "./ssl/ca/{{ scylla_cluster_name }}-ca.crt"
+      register: ca_crt_details
 
     - name: Generate an OpenSSL private key for the CA.
       openssl_privatekey:
         path: "./ssl/ca/{{ scylla_cluster_name }}-ca.pem"
+      when: ca_crt_details.stat.exists == False
 
     - name: Generate an OpenSSL Certificate Signing Request for the CA
       openssl_csr:
         path: "./ssl/ca/{{ scylla_cluster_name }}-ca.csr"
         privatekey_path: "./ssl/ca/{{ scylla_cluster_name }}-ca.pem"
         common_name: "{{ scylla_cluster_name}}.internal"
+      when: ca_crt_details.stat.exists == False
```

Comment:
Why generate a csr if we tested the presence of a certificate. The csr may exist already.

Reply:
Hi sorry for the late reply, I think at the time I was having troubles with my particular setup, I use vault to generate my ca.crt, and machine crt/pem files and when I dropped those files in and ran this plan it overwrote the files I had setup